src/ExpenseTracker.Web/Controllers/TransactionCategoryController.cs

```diff
@@ -121,11 +129,13 @@ await _transactionCategoryService.Update(new TransactionCategoryUpdateDto()
             return RedirectToAction(nameof(Index));
         }
 
-        public async Task<IActionResult> Delete(int transactionCategoryId)
+        public async Task<IActionResult> Delete(long id)
         {
             try
             {
-                var transactionCategory = await _transactionCategoryRepository.FindAsync(transactionCategoryId) ?? throw new TransactionCategoryNotFoundException();
+                var transactionCategory = await _transactionCategoryRepository.FindAsync(id) ?? throw new TransactionCategoryNotFoundException();
+                var defaultWorkspaceId = await GetDefaultWorkspaceId();
+                if (transactionCategory.WorkspaceId != defaultWorkspaceId) return RedirectToAction(nameof(Index));
```

Comment:
@BirajMainali This shows no indication to the user. We can throw an exception here and the catch handler should show the error message.